### PR TITLE
repairs for new networking code

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

File: assignment-client/src/Agent.cpp [hunk 1]
```diff
@@ -183,6 +193,9 @@ void Agent::run() {
 
 void Agent::aboutToFinish() {
     _scriptEngine.stop();
+    
+    auto& packetReceiver = DependencyManager::get<NodeList>()->getPacketReceiver();
+    packetReceiver.unregisterListener(DependencyManager::get<AvatarHashMap>().data());
```
Comment (huffman) [inline assignment-client/src/Agent.cpp]: This can be removed

File: libraries/avatars/src/AvatarHashMap.h [hunk 2]
```diff
@@ -36,14 +35,15 @@ class AvatarHashMap : public QObject, public Dependency, public PacketListener {
...
-
-private slots:
-    void sessionUUIDChanged(const QUuid& sessionUUID, const QUuid& oldUUID);
+    
     void processAvatarDataPacket(QSharedPointer<NLPacket> packet, SharedNodePointer sendingNode);
     void processAvatarIdentityPacket(QSharedPointer<NLPacket> packet, SharedNodePointer sendingNode);
     void processAvatarBillboardPacket(QSharedPointer<NLPacket> packet, SharedNodePointer sendingNode);
     void processKillAvatar(QSharedPointer<NLPacket> packet, SharedNodePointer sendingNode);
 
+private slots:
```
Comment (Atlante45) [inline libraries/avatars/src/AvatarHashMap.h]: Put those slots back to private.

File: libraries/networking/src/LimitedNodeList.h [hunk 3]
```diff
@@ -119,16 +116,19 @@ class LimitedNodeList : public QObject, public Dependency, public PacketListener
 
     bool packetSourceAndHashMatch(const NLPacket& packet, SharedNodePointer& matchingNode);
 
-    PacketReceiver& getPacketReceiver() { return _packetReceiver; }
+    PacketReceiver& getPacketReceiver() { return *_packetReceiver; }
```
Comment (Atlante45) [inline libraries/networking/src/LimitedNodeList.h:119]: This doesn't look safe.

File: libraries/networking/src/PacketListener.cpp [hunk 4]
```diff
@@ -1,18 +0,0 @@
...
-//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
-//
-
-#include "PacketListener.h"
-
-#include "NodeList.h"
-
-PacketListener::~PacketListener() {
-    DependencyManager::get<NodeList>()->getPacketReceiver().unregisterListener(this);
-}
```
Comment (Atlante45) [inline libraries/networking/src/PacketListener.cpp:18]: This looks awfully wrong, did that get deleted during a merge or something?

**Pull request merged**